buildchain/buildchain/codegen.py

```diff
@@ -0,0 +1,49 @@
+# coding: utf-8
+
+
+"""Tasks for code generation."""
+
+
+import shlex
+from typing import Callable, Iterator, Tuple
+
+import doit  # type: ignore
+
+from buildchain import config
+from buildchain import constants
+from buildchain import types
+from buildchain import utils
+
+def task_codegen() -> Iterator[types.TaskDict]:
+    """Run the code generation tools."""
+    for create_codegen_task in CODEGEN:
+        yield create_codegen_task()
+
+
+def codegen_go() -> types.TaskDict:
+    """Generate Go code using operator-sdk."""
+    cwd  = constants.STORAGE_OPERATOR_ROOT
+    actions = []
+    for target in ('k8s', 'openapi'):
```

Comment:
To be really picky, this should be `['k8s', 'openapi']` (since it's a `list<str>` rather than a `(str, str)`), but let's not let that keep this PR unmerged :-P

Reply:
Why? Tuples are lighter than lists

Reply:
I used tuple for its immutability (since the list is kind of a constant here), but in this context tuple and list are really alike: won't make any (observable) difference.

Reply:
The remark came indeed from a 'static typing' background:

- A *list* is an *ordered* collection of an *unknown* number of values of a *single* type
- A *tuple* is an *ordered* collection of a *known* number of values of (potentially) *different* (but known) types

As such, the type constructor of a list has a single argument, whilst the type constructor of an N-tuple has N arguments.

The immutability of a tuple doesn't really play a role in the argument.

Anyway, pedantic, though in general I'm not overly fond of conflating the two (yes, despite duck-typing).